benchmark/acl/blas3/gemm.cpp

```diff
@@ -0,0 +1,163 @@
+      << n;
+  return str.str();
+}
+
+void run(benchmark::State& state, int t1, int t2,
+         index_t m, index_t k, index_t n, float alpha, float beta) {
+  // Standard test setup.
+  std::string t1s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t1));
+  std::string t2s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t2));
+  const char* t_a = t1s.c_str();
+  const char* t_b = t2s.c_str();
+
+  index_t lda = t_a[0] == 'n' ? m : k;
+  index_t ldb = t_b[0] == 'n' ? k : n;
+  index_t ldc = m;
+
+  state.counters["m"] = m;
+  state.counters["k"] = k;
+  state.counters["n"] = n;
+
+  // The counters are double. We convert m, n and k to double to avoid
+  // integer overflows and write them in the counters
+  double m_d = static_cast<double>(m);
+  double n_d = static_cast<double>(n);
+  double k_d = static_cast<double>(k);
+
+  state.counters["n_fl_ops"] = 2 * (m_d * n_d * k_d) + 3 * (m_d * n_d);
+  state.counters["bytes_processed"] =
+      (m_d * k_d + k_d * n_d + 2 * m_d * n_d) * sizeof(float);
+  if (beta == 0.0) {
+    // not adding beta * C
+    state.counters["n_fl_ops"] -= 2 * m_d * n_d;
+    // not reading C
+    state.counters["bytes_processed"] -= m_d * n_d * sizeof(float);
+  }
+
+  // Matrices
+  std::vector<float> a = blas_benchmark::utils::random_data<float>(m * k);
+  std::vector<float> b = blas_benchmark::utils::random_data<float>(k * n);
+  std::vector<float> c =
+      blas_benchmark::utils::const_data<float>(m * n, 0);
+
+  // Device matrices
+  const arm_compute::TensorShape shape_a(k, m), shape_b(n, k), shape_c(n, m);
+#ifdef ACL_BACKEND_NEON
+  arm_compute::Tensor arm_a, arm_b, arm_c;
+#else
+  arm_compute::CLScheduler::get().default_init();
+  arm_compute::CLTensor arm_a, arm_b, arm_c;
+#endif
+  arm_a.allocator()->init(arm_compute::TensorInfo(shape_a, 1, arm_compute::DataType::F32));
+  arm_b.allocator()->init(arm_compute::TensorInfo(shape_b, 1, arm_compute::DataType::F32));
+  arm_c.allocator()->init(arm_compute::TensorInfo(shape_c, 1, arm_compute::DataType::F32));
+  arm_a.allocator()->allocate();
+  arm_b.allocator()->allocate();
+  arm_c.allocator()->allocate();
+  blas_benchmark::utils::fill_tensor(arm_a, a);
+  blas_benchmark::utils::fill_tensor(arm_b, b);
+  blas_benchmark::utils::fill_tensor(arm_c, c);
+
+  // Configure the BLAS routine
+#ifdef ACL_BACKEND_NEON
+  arm_compute::NEGEMM arm_gemm;
+#else
+  arm_compute::CLGEMM arm_gemm;
+#endif
+  arm_gemm.configure(&arm_a, &arm_b, nullptr, &arm_c, alpha, beta);
+
+  // Create a utility lambda describing the blas method that we want to run.
+  auto blas_method_def = [&]() -> std::vector<void*> {
+    arm_gemm.run();
+#ifdef ACL_BACKEND_OPENCL
+    arm_compute::CLScheduler::get().sync();
+#endif
+    return {nullptr};
+  };
+
+  // Warm up to avoid benchmarking data transfer
+  // blas_benchmark::utils::warmup(blas_method_def);
```

Comment:
Is this deliberate to be commented out?

Reply:
It was because the function was moved to the `common_utils` in another MR, now that it's updated on top of everything I enabled the warmup